src/rand.jl

```diff
@@ -0,0 +1,58 @@
+import Random
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    st::Random.SamplerType{TOrRef},
+    n::Random.Repetition,
+) where {TOrRef<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
+    T = _nonreftype(TOrRef)
+    # return Random.SamplerSimple(
+    #     Random.SamplerType{T}(),
+    #     Random.SamplerBigFloat(precision(T)),
+    # )
+    res = setprecision(BigFloat, precision(T)) do
+        Random.SamplerSimple(Random.SamplerType{T}(), Random.Sampler(RNG, BigFloat, n))
+    end
+    return res
+end
+
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arf}) where {A,B} =
+    Arf(rand(rng, sp.data), prec = sp.data.prec)
```

Comment:
Isn't it necessary here also to set the `BigFloat` precision according to `sp` before calling `rand` ?

Reply:
here `sp.data` is `Sampler(RNG, BigFloat, n)` which was created within `setprecision` `do` block`. I understood that this will fix the precision?

Reply:
Yes, this sets the precision of the `Sampler`, but until very recently, there is a bug such that `rand(sp)` will create a `BigFloat` with the current precision instead of the precision of the `sp` (so in particular this will crash sometimes when `sp.prec` is bigger than the current precision). I though you were refering to this bug in the TODOS. So this is fixed in 1.6, if you like I can push the fix to the 1.5 backport branch.

Reply:
```suggestion
    setprecision(BigFloat, sp.data.prec) do 
        Arf(rand(rng, sp.data), prec = sp.data.prec)
    end
```